_sips/minutes/2017-10-24-sip-minutes.md

```diff
@@ -0,0 +1,233 @@
+* automatic typechecking of spliced untyped trees, little explored
+
+These solutions have different trade-offs with regards to
+
+- feature support
+- portability
+- breaking changes with existing macro ecosystem
+
+
+*Generation macros*
+
+There seem to be no roadblockers for supporting macros that do no tree
+inspection, only untyped tree construction.
+Code generation macros cannot change.
+
+Open discussion about the above proposed
+
+[YouTube time: 23'39'' until the end](https://youtu.be/aIc-o1pcRhw?t=1419)
+
+About 40 minutes were dedicated to finding a common ground for the  direction of the project as well as the technical details that should be addressed going forward.
+
+**Adriaan** immediately pointed out that by taking on the "hard stuff" to deal with first will give clarity to what could be supported for the next year. He suggests tackling the more ambitious features first, by experimenting in a current macro system, is a "cheap way" to find out the "unknown". The goal would be to find out how to implement it in the next macro system.
+Challenges that can be taken on:
+[(https://youtu.be/aIc-o1pcRhw?t=1673)]
+- splicing trees, ending up with owner chains that are correct
+- hygiene
+- better tools for macro authors, to experiment in a current macro system
+
+**Eugene**
+https://youtu.be/aIc-o1pcRhw?t=1877
+Agrees both with **Adriaan** and **Olaf**, on one hand "prototyping" in the current macro system can be beneficial e.g automatic owner chain fixer but on the other, scala reflect is fundamentally different from the prototype macro system and even if the tests are run there might be no point to it.
+He does agree with **Olaf** that classification of the macros is useful. Concluding that supporting the generation transformation macros shouldn't be that hard, but than the question of how valuable to the community these are needs to be raised.
+-- OLAF NEEDS TO RE-WRITE THIS PART --
+
+Other question raised:
+
+- discuss the quasiquote (by **Martin**, **Olaf** and **Heather**)
+ Go to: https://youtu.be/aIc-o1pcRhw?t=1960
+ "How restrictive is to demand all quasiquote to be fully typed instead of free name reference to type up the abstract" **Martin**
+-- PLEASE SUMMARISE THE MAIN POINTS IN THIS DISCUSSION --
+
+ - White box / Black box(by **Martin** / **Olaf**)
+  Main concern "when do tools have to run the untrusted code?"
+
+ - Annotation macros (Community)
+ **Martin** insists on leaving the paradise functionality in "meta/template scala" but that *scala* assumes only the things that have proven themselves e.g. simulacrum
+ -- PLEASE ADD THE REASON IF NECESSARY
+
+ He strongly believes that macro annotations are a "complete abuse of macros" and will make sure that "this things are not possible anymore"
+
+- The interaction between type inference and macros (**Iulian**)
+quote https://youtu.be/aIc-o1pcRhw?t=3436
+-- OLAF, PLEASE MAKE SURE THE SENSE IS MADE OF WHAT IULIAN SAID --
+
```

Comment:
@olafurpg

Reply:
> **Iulian** would like to see addressed the interaction between type inference and macros. This interaction seems to be one of the hard unsolved problems in the current macro system. For example, code that uses shapeless typically has two type parameters where one parameter is unbounded and it's inferred type is guided by the shapeless macro.

Reply:
@olafurpg THANK YOU! (this goes for all the comments below)